domain/src/main/java/com/github/dodobest/domain/Calculator.kt

```diff
@@ -1,186 +1,168 @@
 package com.github.dodobest.domain
 
-class Calculator {
+enum class Operation (
+    private val operation: String
+) {
+    PLUS("+"),
+    MINUS("-"),
+    MULTIPLY("*"),
+    DIVIDE("/");
```

Comment:
```suggestion
enum class Operator(
    private val symbol: String,
    private val calculateStrategy: (Int, Int) -> Int
) {
    PLUS("+", { left, right -> left + right }),
    MINUS("-", { left, right -> left - right }),
    MULTIPLY("*", { left, right -> left * right }),
    DIVIDE("/", { left, right -> left / right });
    
    fun calculate(left: Int, right: Int): Int {
        return calculateStrategy(left, right)
    }
    
    companion object {
        fun find(symbol: String): Operator = values().find { it.symbol== symbol}
            ?: throw IllegalArgumentException("$symbol을 가진 연산자를 찾을 수 없습니다.")
    }
```
```kotlin
val operator = Operator.find("+")
val result = operator.calculate(1, 2)
// result == 3
```

이런식으로 계산 전략에 대한 람다를 직접 Enum의 멤버 변수로써 갖는다면, when이나 if문 없이도 바로 계산이 가능하게 됩니다.